### ISSUE TITLE
Improve coverage of the unit tests

### ISSUE THREAD

**Issue by adonath** [xpsi-group/neost#62]
I locally created a coverage report based on the existing tests with the following output:

![Screenshot 2024-11-14 at 10 34 28 AM](https://github.com/user-attachments/assets/2a474946-df22-4830-9d52-4f8bda077e3f)

I think it would be good to try to improve the coverage a bit, so add a few more test cases to make sure a larger part of the code base is actually executed during the tests. The report can be re-created (from the base folder) using:

```python
python -m pip install coverage
```

```
coverage run -m pytest tests
coverage html && open htmlcov/index.html
```

There is no formal requirement on the coverage value by JOSS, it just asks for the existence of any tests. But it seems a larger part of the core functionality (e.g. in `likelihood.py` is never executed). Additionally I can recommend to have the example executed as part of the CI system. If the examples run too long, maybe just reduce the sampling size. This verifies they run at all.

**Comment by svisak** [comment]
Agreed, this is very similar to issue #32. The unit tests don't have to be completely comprehensive, but they should ensure that all core functionality works as expected. I think the full developer team should discuss this.
